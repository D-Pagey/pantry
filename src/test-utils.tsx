import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC, ReactNode } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

type AllTheProvidersProps = {
    children: ReactNode;
    customContext?: any;
};

const AllTheProviders: FC<AllTheProvidersProps> = ({ children, customContext }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            },
            ...customContext
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, customContext?: any, options?: any) =>
    render(ui, { wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
