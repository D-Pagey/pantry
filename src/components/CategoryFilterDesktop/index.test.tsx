import React from 'react';
import { render } from '../../test-utils';
import { CategoryFilterDesktop } from '.';

const props = {
    categories: { meat: 5, fish: 2, vegetables: 1, dairy: 1 },
    handleCategoryClick: (category: string) => console.log({ category }),
    selected: 'all'
};

describe('CategoryFilterDesktop component', () => {
    it('should render', () => {
        const { container } = render(<CategoryFilterDesktop {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
