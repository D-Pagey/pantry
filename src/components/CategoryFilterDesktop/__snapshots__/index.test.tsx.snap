// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilterDesktop component should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  font-size: 26px;
  margin: 0;
  padding: 0;
}

.c1 {
  border-bottom: 2px solid;
  border-color: #348841;
  color: #348841;
  cursor: pointer;
  padding: 0 1rem 1rem;
}

.c1:hover {
  color: #348841;
  border-bottom: 2px solid #348841;
}

.c1:first-child {
  padding: 0 0.5rem 1rem;
}

.c2 {
  border-bottom: 2px solid;
  cursor: pointer;
  padding: 0 1rem 1rem;
}

.c2:hover {
  color: #348841;
  border-bottom: 2px solid #348841;
}

.c2:first-child {
  padding: 0 0.5rem 1rem;
}

<ul
  class="c0"
>
  <li
    class="c1"
  >
    All
    (
    5
    )
  </li>
  <li
    class="c1"
  >
    All
    (
    10
    )
  </li>
  <li
    class="c2"
  >
    Meat
    (
    5
    )
  </li>
  <li
    class="c2"
  >
    Fish
    (
    2
    )
  </li>
  <li
    class="c2"
  >
    Vegetables
    (
    1
    )
  </li>
  <li
    class="c2"
  >
    Dairy
    (
    1
    )
  </li>
</ul>
`;
